src/public/device.ts

```diff
@@ -0,0 +1,93 @@
+import { GlobalVars } from '../internal/globalVars';
+import { ensureInitialized, sendMessageRequestToParent } from '../internal/internalAPIs';
+import { frameContexts } from '../internal/constants';
+
+/**
+ * Namespace for device related APIs. These are implemented only for mobile (Android and iOS)
+ * On desktop, these APIs will be a noop.
+ */
+export namespace device {
+  /**
+   * Status codes for device APIs
+   * @see File
+   */
+  export enum StatusCode {
+    Success = 0,
+    /**
+     * Missing required permission to perform the action
+     */
+    PermissionError = 1,
+    /**
+     * Error encountered while performing the required operation.
+     * e.g. Camera failed, compression failed etc
+     */
+    InternalError = 2,
+    /**
+     * User cancelled the action
+     */
+    UserCancelled = 3,
+  }
+
+  /**
+   * Enum for file formats supported
+   */
+  export enum FileFormat {
+    Base64 = 'base64',
+  }
+
+  /**
+   * File object that can be used to represent image or video or audio
+   * The object will have "statusCode" as defined in StatusCode enum
+   * The user of this API should check "statusCode" to be "Success" before reading the "content"
+   * In case of other status codes, app can decide to show an error message to the user
+   * @see StatusCode
+   */
+  export interface File {
+    /**
+     * Status code
+     */
+    statusCode: StatusCode;
```

Comment:
The error is associated with a File because if user selected more than one file and if we are able to return one file and we encountered problem while compressing / processing other file, it will be error per File.Returning a top level error and not responding with partial response is another option, which will move this error out to top level.

Reply:
Modified signature to have the callback as (error, file[])